src/components/Posts/EditPostModal.tsx

```diff
@@ -0,0 +1,190 @@
+  );
+
+  useEffect(() => {
+    setPostTextContent(originalPostTextContent);
+  }, [originalPostTextContent]);
+
+  const handleEdit = async () => {
+    try {
+      await axios.put(
+        "http://localhost:3001/api/posts/updatePost",
+        {
+          postId: postId,
+          textContent: postTextContent,
+          editedTimestamp: new Date(),
+        },
+        {
+          headers: {
+            "Content-Type": "application/json",
+          },
+        }
+      );
+
+      if (existingPost) {
+        dispatch(
+          updatePost({
+            ...existingPost,
+            textContent: postTextContent,
+            editedTimestamp: new Date().toISOString(),
+          })
+        );
+      }
+      onClose?.();
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  return (
+    <Dialog
+      fullWidth
+      onClose={onClose}
+      open={openModal}
+      PaperProps={{ style: styles.paperProps }}
+      scroll="paper"
+      sx={styles.dialog}
+    >
+      <DialogTitle sx={styles.dialogTitle}>
+        <IconButton onClick={onClose}>
+          <CloseIcon />
+        </IconButton>
+        <Typography sx={styles.titleText}>Edit Post</Typography>
+      </DialogTitle>
+      <DialogContent>
+        <Box sx={styles.dialogContentContainer}>
+          <Box sx={styles.editPostHeader}>
+            <Box sx={styles.avatar}>
+              <UserAvatar username={user.username} />
+            </Box>
+            <Box>
+              <Box sx={styles.userInfoText}>
+                <Typography variant="subtitle1">{user.username}</Typography>
+                <Typography variant="subtitle2">{`@${user.displayName}`}</Typography>
+              </Box>
+              <Typography variant="body2">
+                {prevEditedPostTimestamp
+                  ? formatTimestamp(prevEditedPostTimestamp, true)
+                  : formatTimestamp(originalPostTimeStamp)}
+              </Typography>
+            </Box>
+          </Box>
+          <Box sx={styles.textFieldContainer}>
+            <TextField
+              fullWidth
+              hiddenLabel
+              multiline
+              onChange={(e) => setPostTextContent(e.target.value)}
+              sx={styles.textField}
+              value={postTextContent}
+              variant="standard"
+            />
+            <Box sx={styles.postActions}>
+              <Stack direction="row">
+                <IconButton size="small">
+                  <AddPhotoAlternateOutlinedIcon />
+                </IconButton>
+                <EmojiPickerIconButton
+                  onEmojiClick={(emoji: EmojiClickData) => {
+                    setPostTextContent(
+                      (prevContent) => prevContent + emoji.emoji
+                    );
+                  }}
+                />
+              </Stack>
+              <Button
+                size="small"
+                type="submit"
+                variant="contained"
+                onClick={handleEdit}
```

Comment:
For consistency let's wrap the entire thing in a `form` and make this into a submit button. I don't think it makes a huge difference but since we're using a from in our other dialogs let's follow the same pattern here. The button actually already has `type="submit"` attribute on it, probably from a copy paste somewhere lol

If you do this you'll remove the `onClick` and assign `handleEdit` to the `onSubmit` of the form instead

Reply:
done